google/cloud/storage/internal/grpc/metrics_exporter_options_test.cc

```diff
@@ -67,54 +83,128 @@ TEST(MetricsExporterOptions, DefaultEndpoint) {
+    EXPECT_TRUE(actual.get<otel_internal::ServiceTimeSeriesOption>());
+    EXPECT_EQ(actual.get<otel_internal::MetricPrefixOption>(),
+              "storage.googleapis.com/");
   }
 }
 
 TEST(MetricsExporterOptions, PrivateDefaultUD) {
   for (std::string prefix : {"", "google-c2p:///"}) {
     SCOPED_TRACE("Testing with prefix = " + prefix);
     auto actual = MetricsExporterOptions(
+        Project("test-only"), EmptyResource(),
         Options{}.set<EndpointOption>(prefix + "private.googleapis.com"));
     EXPECT_THAT(actual.get<EndpointOption>(), "private.googleapis.com");
+    EXPECT_TRUE(actual.has<otel_internal::MonitoredResourceOption>());
+    EXPECT_TRUE(actual.get<otel_internal::ServiceTimeSeriesOption>());
+    EXPECT_EQ(actual.get<otel_internal::MetricPrefixOption>(),
+              "storage.googleapis.com/");
   }
 }
 
 TEST(MetricsExporterOptions, PrivateUD) {
   for (std::string prefix : {"", "google-c2p:///"}) {
     SCOPED_TRACE("Testing with prefix = " + prefix);
     auto actual = MetricsExporterOptions(
+        Project("test-only"), EmptyResource(),
         Options{}
             .set<EndpointOption>(prefix + "private.ud.net")
             .set<internal::UniverseDomainOption>("ud.net"));
     EXPECT_THAT(actual.get<EndpointOption>(), "private.ud.net");
+    EXPECT_TRUE(actual.has<otel_internal::MonitoredResourceOption>());
+    EXPECT_TRUE(actual.get<otel_internal::ServiceTimeSeriesOption>());
+    EXPECT_EQ(actual.get<otel_internal::MetricPrefixOption>(),
+              "storage.googleapis.com/");
   }
 }
 
 TEST(MetricsExporterOptions, RestrictedDefaultUD) {
   for (std::string prefix : {"", "google-c2p:///"}) {
     SCOPED_TRACE("Testing with prefix = " + prefix);
     auto actual = MetricsExporterOptions(
+        Project("test-only"), EmptyResource(),
         Options{}.set<EndpointOption>(prefix + "restricted.googleapis.com"));
     EXPECT_THAT(actual.get<EndpointOption>(), "restricted.googleapis.com");
+    EXPECT_TRUE(actual.has<otel_internal::MonitoredResourceOption>());
+    EXPECT_TRUE(actual.get<otel_internal::ServiceTimeSeriesOption>());
+    EXPECT_EQ(actual.get<otel_internal::MetricPrefixOption>(),
+              "storage.googleapis.com/");
   }
 }
 
 TEST(MetricsExporterOptions, RestrictedUD) {
   for (std::string prefix : {"", "google-c2p:///"}) {
     SCOPED_TRACE("Testing with prefix = " + prefix);
     auto actual = MetricsExporterOptions(
+        Project("test-only"), EmptyResource(),
         Options{}
             .set<EndpointOption>(prefix + "restricted.ud.net")
             .set<internal::UniverseDomainOption>("ud.net"));
     EXPECT_THAT(actual.get<EndpointOption>(), "restricted.ud.net");
+    EXPECT_TRUE(actual.has<otel_internal::MonitoredResourceOption>());
+    EXPECT_TRUE(actual.get<otel_internal::ServiceTimeSeriesOption>());
+    EXPECT_EQ(actual.get<otel_internal::MetricPrefixOption>(),
+              "storage.googleapis.com/");
   }
 }
 
+auto IsInstanceId() {
+  return ContainsRegex(
```

Comment:
Prior art says this can be simplified: https://github.com/googleapis/google-cloud-cpp/blob/dc92806e6190a235a23d64d1d7bbb0d4d5425b81/generator/integration_tests/tests/request_id_connection_impl_test.cc#L64-L68

Reply:
That prior art depends on complexifying other parts of the code:

https://github.com/googleapis/google-cloud-cpp/blob/dc92806e6190a235a23d64d1d7bbb0d4d5425b81/generator/integration_tests/tests/request_id_connection_impl_test.cc#L90-L92

Reply:
(I am not asking for changes, just FYI)

There is also prior art on avoiding `::testing::*Regex`

https://github.com/googleapis/google-cloud-cpp/blob/dc92806e6190a235a23d64d1d7bbb0d4d5425b81/google/cloud/testing_util/validate_metadata.cc#L102-L113

Reply:
Done

Reply:
Nice. Thanks.